public/requests.js

```diff
@@ -0,0 +1,109 @@
+import md5_encode from './MD5.js'
+
+/**
+ *  добавление eventListener на кнопку поиска трека
+ */
+const search_input = document.querySelector('.header__search');
+search_input.addEventListener('keydown', function(e) {
+    if (e.keyCode === 13) {
+        send_request('GET','track.search',`&track=${search_input.value}`,search_input.value).then(
+            res => get_content('search',res.results.trackmatches.track)
+        )
+    }
+});
+
+/**
+ *  добавление eventListener на кнопку любимых треков
+ */
+const fav_button = document.querySelector('.fav');
+fav_button.addEventListener('click', function() {
+    send_request('GET','user.getlovedtracks','&user=Foxyb0y').then(
+        res => get_content('favorites',res.lovedtracks.track)
+    )
+});
+
+/**
+ *
+ * @param method метод Апи (GET или POST)
+ * @param request_method параметр запроса - метод
+ * @param require_param параметр определяет какой GET-запрос отправить
+ * @param track название трека
+ * @param artist имя артиста
+ * @returns {Promise} Выполненный промис
+ */
+async function send_request(method,request_method,require_param='',track= '',artist='') {
+    const url = 'http://ws.audioscrobbler.com/2.0/'
+    const api_key = '906db58ae0258689ba249d53210358ee';
+    try {
+        if (method === 'GET') {
+            const response = await fetch(`${url}?api_key=${api_key}&method=${request_method}${require_param}&format=json`, {
+                method: 'GET'
+            });
+            return await response.json()
+        } else if (method === 'POST') {
+            const sk = 'PEJSZfVNq8UlcfpQoiME1HzKEaFYAClf';
+            const secret = '04e5fe9b4835a8387149c2770345a9af';
+            const params = new URLSearchParams({
+                'track': track, 'artist': artist, 'api_key': api_key,
+                'api_sig': `${md5_encode(api_key, artist, sk, track, secret)}`, 'sk': sk, 'method': request_method
+            });
+            const response = fetch(url, {
+                method: 'POST',
+                body: params,
+                headers: {'Content-Type': 'application/x-www-form-urlencoded'}
+            })
+            return await response.json();
+        }
+    } catch (err) {
+        alert(`Произошла ошибка ${err}`)
```

Comment:
так выведется не то. что ожидалось.
Должно быть как-то так
```suggestion
        alert(`Произошла ошибка ${err.message}`)
```
Так же при возникновении стоит кидать исключение, чтобы можно было избежать случая, который я описал выше